google/cloud/internal/async_polling_loop.cc

```diff
@@ -45,8 +45,12 @@ class AsyncPollingLoopImpl
   future<StatusOr<Operation>> Start(future<StatusOr<Operation>> op) {
     auto self = shared_from_this();
     auto w = WeakFromThis();
-    promise_ = promise<StatusOr<Operation>>([w] {
-      if (auto self = w.lock()) self->DoCancel();
+    auto const& options = CurrentOptions();
+    promise_ = promise<StatusOr<Operation>>([w, options]() mutable {
+      if (auto self = w.lock()) {
+        OptionsSpan span(std::move(options));
+        self->DoCancel();
+      }
     });
     op.then([self](future<StatusOr<Operation>> f) { self->OnStart(f.get()); });
```

Comment:
I think we want `OnStart()` to run with the options in effect during `Start()`?  If so, it seems that we want to capture `CurrentOptions()` in the constructor of this class, and create a span with them on each `OnFoo()` callback

Reply:
See for example lines 79, and 123.

Reply:
Aahhh, I forgot about those.